lntest/itest/lnd_payment_test.go

```diff
@@ -374,19 +375,36 @@ func testAsyncPayments(net *lntest.NetworkHarness, t *harnessTest) {
 		}
 
 		return true
-	}, time.Second*5)
+	}, defaultTimeout)
 	if err != nil {
 		t.Fatalf("failed to assert alice's pending htlcs and/or remote/local balance")
 	}
 
 	// Wait for Bob to receive revocation from Alice.
-	time.Sleep(2 * time.Second)
+	var bobChan *lnrpc.Channel
+	err = wait.Predicate(func() bool {
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		bobChan, err = getChanInfo(ctxt, net.Bob)
+		if err != nil {
+			t.Fatalf("unable to get bob's channel info: %v", err)
+		}
+
+		if len(bobChan.PendingHtlcs) != 0 {
+			return false
```

Comment:
nit: could use `wait.NoError()` instead so we'd see which condition wasn't satisfied in the end. But non-blocking as this predicate shouldn't time out in the first place.

Reply:
The diff is not super good in this regard but I kept the error displays intact which are right after the `Predicate` block.

Reply:
I'll update to use `wait.NoError()`, it's much nicer.

Reply:
okay, now this is fixed.